packages/environment-ember-loose/package.json

```diff
@@ -28,6 +28,7 @@
     "registry/**/*.{js,d.ts}"
   ],
   "peerDependencies": {
+    "@ember/test-helpers": "^2.9.3",
```

Comment:
This should probably be optional for the same reasons as most of the rest of these are, no? Essentially if you have `@ember/test-helpers` installed, we want it as a peer. If you don't, then that's ok, we don't _rely_ on it, we just merge types for it.

Reply:
That's not quite the case, due to this import: https://github.com/typed-ember/glint/blob/11a69e8e835329ef88dd77ee4820e85ef0b760ef/packages/environment-ember-loose/-private/dsl/integration-declarations.d.ts#L102

The module must be defined before it can be imported.

Reply:
They all do that. Side-effect imports never produce a type error.

Reply:
@NullVoxPopuli if you'll fix this I'd love to get it included in the next beta release

Reply:
Yeah, that’s because there’s nothing special about Glint there! It’s just TypeScript. The thing to get sorted out is that you need to get rid of all copies of `@types/ember__test-helpers` and switch over to just using the types from `@ember/test-helpers`. And yeah, that's an annoying experience, but it's also one that no one but TS itself has any control over.

Reply:
Where do we stand on this @NullVoxPopuli?

Reply:
I haven't tried making a project with glint, yet without the test-helpers in a while.

I need to verify the behavior at the top of this thread

Reply:
I have a reproduction atm -- and this trace:
```
======== Resolving type reference directive 'ember__test-helpers', containing file '<repo>/ember-resources/__inferred type names__.ts', root directory '<repo>/ember-resources/node_modules/@types,<repo>/node_modules/@types'. ========
Resolving with primary search path '<repo>/ember-resources/node_modules/@types, <repo>/node_modules/@types'.
Found 'package.json' at '<repo>/ember-resources/node_modules/@types/ember__test-helpers/package.json'.
'package.json' does not have a 'typesVersions' field.
'package.json' does not have a 'typings' field.
'package.json' does not have a 'types' field.
File '<repo>/ember-resources/node_modules/@types/ember__test-helpers/index.d.ts' does not exist.
Looking up in 'node_modules' folder, initial location '<repo>/ember-resources'.
File '<repo>/ember-resources/node_modules/ember__test-helpers.d.ts' does not exist.
File '<repo>/ember-resources/node_modules/@types/ember__test-helpers/package.json' exists according to earlier cached lookups.
File '<repo>/ember-resources/node_modules/@types/ember__test-helpers.d.ts' does not exist.
'package.json' does not have a 'typings' field.
'package.json' does not have a 'types' field.
File '<repo>/ember-resources/node_modules/@types/ember__test-helpers/index.d.ts' does not exist.
File '<repo>/node_modules/ember__test-helpers.d.ts' does not exist.
File '<repo>/node_modules/@types/ember__test-helpers.d.ts' does not exist.
File '/✂️/node_modules/ember__test-helpers.d.ts' does not exist.
Directory '/✂️/node_modules/@types' does not exist, skipping all lookups in it.
Directory '/✂️/node_modules' does not exist, skipping all lookups in it.
Directory '🏠/node_modules' does not exist, skipping all lookups in it.
Directory '/home/node_modules' does not exist, skipping all lookups in it.
Directory '/node_modules' does not exist, skipping all lookups in it.
======== Type reference directive 'ember__test-helpers' was not resolved. ========}
```

Reply:
_deleting_ `@types/ember__test-helpers` resolved the error.

I'm about to add `@types` and native ember types testing to C.I., hopefully, :crossed_fingers: so I hope I can get both styles of type-providing tested/verified simultaneously via matrix

Reply:
Removing `@ember/test-helpers` altogether (both types and real package) has tsc happy -- so it appears that was the real solution all along -- having the `@types` package caused the above error, having both the `@types` and the real package didn't change anything, but I don't need these packages present so :shrug: